File: 8-dynamic/apis/parser.go
package main

import (
	"net"
	"net/http"
	"strings"
)

type outputParser struct {
	IP       string `json:"Ip Address"`
	Language string `json:"Language"`
	OS       string `json:"Operative System"`
}

func parser(w http.ResponseWriter, r *http.Request) {
	param := strings.Split(r.URL.Path, "/")
	var data outputParser
	if strings.ToLower(param[3]) == "whoami" {
		data.IP = getIP(r)
		data.Language = r.Header.Get("Accept-Language")
		data.OS = r.Header.Get("User-Agent")
		sendStructAsJSON(w, r, data)
	} else {
		http.Redirect(w, r, "/parser/parser.html", 301)
	}
}

func getIP(r *http.Request) string {
	ip := r.Header.Get("X-Forwarded-For")
	if len(ip) > 0 {
		return ip
	}
	ip, _, _ = net.SplitHostPort(r.RemoteAddr)
	return ip
}

File: old-v1/3-backEnd/apis/nodejs/parser/parser.js
'use strict';
/* jshint node: true */

var express = require('express');
var logger = require('morgan');
var path = require('path');

var app = express();

app.use(logger('dev'));

var publicPath = path.resolve(__dirname, 'public');
app.use(express.static(publicPath));

app.get('/', function (req, res) {
  res.sendFile(publicPath + '/parser.html');
});

app.get('/v1/whoami', function (req, res) {
  var ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress;
  if (ip.substr(0, 7) == '::ffff:') {
    ip = ip.substr(7);
  }
  var lang = (JSON.stringify(req.headers['accept-language'])).slice(1, 6);
  var ops = JSON.stringify(req.headers['user-agent']).split('(');
  ops = ops[1].toString().split(';');
  res.json({
    'IP Address': ip,
    'Language': lang,
    'Operative System': ops[0] + ops[1]
  });
});

app.listen(3002, function () {
  console.log('App started on port 3002');
});
